### PR TITLE
Add support for pluggable discovery

### PR DESCRIPTION
**Pull request by silvanocerza**
This PR adds supports for the new upload syntax following the pluggable discovery specification.

For more info on pluggable discovery: https://arduino.github.io/arduino-cli/0.19/platform-specification/#pluggable-discovery

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -25,6 +27,8 @@ uno2018.build.extra_flags={build.328emulation} -DMILLIS_USE_TIMERB3
 #uno2018.build.extra_flags=-B{runtime.tools.atpack.path}/gcc/dev/{build.mcu}
 
 uno2018.bootloader.tool=avrdude
+uno2018.bootloader.tool.serial=avrdude
+uno2018.bootloader.tool.network=arduino_ota
```
Comment (facchinm) [inline boards.txt:31]: I'm quite sure we don't want to update the bootloader via OTA :) Should we drop this line to explicitly provoke an error?

**Review by facchinm — CHANGES_REQUESTED** [review]

File: boards.txt (hunk 1 above)
Comment (silvanocerza) [reply boards.txt:31]: I don't know really. Arduino CLI technically supports it but I have no idea if it's feasible or not.

Comment (facchinm) [reply boards.txt:31]: @cmaglie

Comment (cmaglie) [reply boards.txt:31]: No, it's not possible because we don't have a `tools.arduino_ota.bootloader.pattern` in platform.txt

**Pull request merged**